membership_group/models/membership_vote_snapshot.py

```diff
@@ -0,0 +1,36 @@
+from odoo import api, fields, models
+
+
+class MembershipVoteSnapshot(models.Model):
+    _name = "membership.vote.snapshot"
+    _description = "Membership Vote Snapshot"
+    rec_name = "partner_id"
+
+    partner_id = fields.Many2one(
+        "res.partner",
+        required=True,
+        readonly=True,
+        ondelete="restrict",
+    )
+    date = fields.Date(
+        required=True,
+        readonly=True,
+    )
+
+    def action_voting_at_date(self):
+        return {
+            "res_model": "membership.vote.history",
+            "views": [[False, "form"]],
+            "target": "new",
+            "type": "ir.actions.act_window",
+        }
+
+    @api.model
+    def action_create_can_vote_snapshot(self):
+        voting_members = self.env["res.partner"].search(
+            [("member_can_vote", "=", True)]
```

Comment:
When I have voting members in the future or in the past it doesn't work. Is it maybe easier to have a domain which uses date_from and date_end / date_to on membership.group.member? And then in membership.vote.history.open_at_date set the selected date as context variable.
Or maybe make member_can_vote dependent on context (api.depends_context)